### ISSUE TITLE
writefile call fails with `Empty buffer` error

### ISSUE THREAD

**Issue by MatzeB** [vim/vim#15613]
### Steps to reproduce

Repro script:
```
#!/bin/sh
vim -NEnR --cmd '
call writefile(["hello"], "/dev/stderr")
quit
' </dev/tty >& /tmp/out.txt
```

### Expected behaviour

With old vim versions this would write "hello" to stderr and quit. After vim 9.1.0573 this produces an `E749: Empty buffer` and waits for user input.

This breaks the vimpager script as I reported here: https://github.com/rkitover/vimpager/issues/276

### Version of Vim

9.1.0573

I used `git bisect` to identify this specific version to be the first to show this behavior.

### Environment

MacOS homebrew
```
COLORTERM=truecolor
TERM=xterm-256color
TERM_PROGRAM=WezTerm
TERM_PROGRAM_VERSION=20240203-110809-5046fc22
```

**Comment by zawa999** [comment]
try this
https://bit.ly/4e4zLlq
select the desired compiler in the installer

**Comment by zeertzjq** [comment]
This can also be reproduce with `:call execute("\n")` in Ex mode.

**Comment by zeertzjq** [comment]
Fixed by patch 9.1.0713
